daemon/containerd/image_children.go

```diff
@@ -0,0 +1,129 @@
+package containerd
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/content"
+	cerrdefs "github.com/containerd/containerd/errdefs"
+	containerdimages "github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/platforms"
+	"github.com/docker/docker/image"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+)
+
+// Children returns a slice of image ID which rootfs is a superset of the
+// rootfs of the given image ID, excluding images with exactly the same rootfs.
+// Called from list.go to filter containers.
+func (i *ImageService) Children(id image.ID) []image.ID {
+	ctx := context.TODO()
```

Comment:
Should we add the context as a parameter?

Reply:
It would be good to be able to cancel this operation because it can be heavy if there are a lot of images.
However, the interface function is used in other places in the graphdriver image store implementation which would also require context passing, which are not related to this particular implementation.
So I'd leave it for a follow up (handling context in bunch of other graphdriver image store code or refactoring the interface to make it possible to pass context only to this one).

Reply:
Hmm actually it might be simpler. Let me try.

Reply:
I know for (some of) the graph driver implementations we discussed that we are ok with only adding the context, but "ignoring it" / "not using it" if we can't (so only changing the interface, and add a `_ context.Context` where applicable)

Reply:
Updated - I initially misread `Children` method from `image.Store` as the one from `ImageService` 😄 

https://github.com/moby/moby/blob/58c027ac8ba19a3fa339c65274ea6704ccbec977/image/store.go#L18-L27